services/skipchain/api.go

```diff
@@ -0,0 +1,130 @@
+package skipchain
+
+import (
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// Client is a structure to communicate with the Skipchain
+// service from the outside
+type Client struct {
+	*sda.Client
+}
+
+// NewClient instantiates a new client with name 'n'
+func NewClient() *Client {
+	return &Client{Client: sda.NewClient("Skipchain")}
+}
+
+// CreateRootInter creates two Skipchains: a root SkipChain with
+// maximumHeight of maxHRoot and an intermediate SkipChain with
+// maximumHeight of maxHInter. It connects both chains for later
+// reference.
+func (c *Client) CreateRootInter(elRoot, elInter *sda.EntityList, maxHRoot, maxHInter int, ver VerifierID) (root, inter *SkipBlock, err error) {
```

Comment:
Yes I got the terminology wrong, it's `Propose{Data,Roster}`, right.
> The CreateRootInter and CreateData are convenience API-calls

I get it, because this is what we needed right now, but you directly jumped from nothing to these convenience calls instead of building smaller blocks (creating simply one skipchain, adding a simple relationship between two skipchains etc) to get to a `CreateRootInter` etc. I'm not saying it does not have to be there, I'm more saying that it should be decomposed because I'm not sure about your assumption 
> because thy will probably be used in most of the cases

Will it always be true ? We don't know (and the way I understood CONIKS with skipchain it did not use 3-levels but I may be wrong) so I think it's best to build small building blocks and create "convenience" calls on top of it. Exactly how we do our helpers like "TreeNodeInstance" in `lib/sda`, a ProtocolInstance developer can do all the hard work it self if it wants to, or just register easy handlers functions.
**Again**, it's not something to do here, more a thought on the whole design in general that could be good to think about for a next iteration.